skylarktest/skylarktest.go

```diff
@@ -134,9 +134,10 @@ func freeze(thread *skylark.Thread, _ *skylark.Builtin, args skylark.Tuple, kwar
 }
 
 // DataFile returns the effective filename of the specified
-// test data resource.  The function abstracts differences between
-// 'go build', under which a test runs in its package directory,
-// and Blaze, under which a test runs in the root of the tree.
-var DataFile = func(pkgdir, filename string) string {
+// test data resource. The symbol allows for overriding of the
+// function used to build file paths. Default bound function
+// processes go build paths.
+var DataFile = func(relDir, filename string) string {
+	pkgdir := "skylark/" + relDir
```

Comment:
Let's fold pkgdir in to the next line, and rename relDir to subdir:
```
return filepath.Join(build.Default.GOPATH, "src/github.com/google/skylark", subdir, filename)
```

FWIW, the implementation we use under Blaze looks like:
```
return filepath.Join(runfiles.Path("google3/third_party/golang"), pkgdir, filename)
```
which means all three implementations apply filepath.Join to the two arguments, so in hindsight only one would have sufficed.

Reply:
Will do. Side note is that this only works for a naive Bazel usage. There are at least 2 permutations of Bazel/rules_go flags, which are very likely to break even my current Bazel support.